packages/util/src/bytes.ts

```diff
@@ -107,8 +106,8 @@ const stripZeros = function (a: any): Buffer | number[] | string {
  * @param a (Buffer)
  * @return (Buffer)
  */
-export const unpadBuffer = function (a: Buffer): Buffer {
-  assertIsBuffer(a)
+export const unpadBuffer = function (a: Uint8Array): Buffer {
```

Comment:
I have again this naming problem here. Additionally I don't think it's a good thing for conversion functions to change the type in between, so here, to then have `Uint8Array` as input and return a `Buffer`. This is also hard to deprecate at some point.

I wouldn't want to do too quick decisions on these kind of things, since implications on how we can deal with this later on can be significant.

So I think if we do we should likely go the a bit harder way and also add here parallel `unpadUint8Array()` and the like functionality and then switch over one-by-one at some point deprecate (in the sense of: write a deprecation note, so not directly remove) the Buffer functions (if we decide that we want that. We can also have both in parallel).